### PR TITLE
fix: re-enable ptpl and allow for not using reservations

### PR DESCRIPTION
**Pull request by tiagolobocastro**
chore: allow for not using reservations

This gives back control to the api client.
To do this, set the reservation parameters to "0".

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

fix(nvme-resv): setup ptpl from the caller side

Previously this was done incorrectly only when props was None.
It's better to leave it for the caller to decide.

### CODE REVIEW

File: io-engine/src/bdev/nexus/nexus_bdev.rs [hunk 1]
```diff
@@ -211,6 +211,12 @@ impl NexusNvmeParams {
     pub fn set_preempt_policy(&mut self, preempt_policy: NexusNvmePreemption) {
         self.preempt_policy = preempt_policy;
     }
+    /// Check if reservations are disabled.
+    pub fn disabled(&self) -> bool {
```
Comment (mtzaurus) [inline io-engine/src/bdev/nexus/nexus_bdev.rs]: Can we use here a name like 'reservations_disabled()' to denote that it deals with reservations ?
Also, as a nit, 'enabled()' looks a bit more natural to me than '.disabled()', but it just a suggestion here.

**Comment by mtzaurus** [comment]
Can we use here a name like 'reservations_disabled()' to denote that it deals with reservations ?
Also, as a nit, 'enabled()' looks a bit more natural to me than '.disabled()', but it just a suggestion here.

**Review by mtzaurus — APPROVED** [review]

File: io-engine/src/bdev/nexus/nexus_bdev.rs (hunk 1 above)
Comment (tiagolobocastro) [reply io-engine/src/bdev/nexus/nexus_bdev.rs]: Ah I thought the type name here was NvmeReservations which is why I left out reservations but looks like it's only NvmeParams, so I'll rename to reservations_enabled().

**Review by chriswldenyer — APPROVED** [review]
Looks reasonable

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**